src/mds/Locker.cc

```diff
@@ -2298,7 +2298,7 @@ uint64_t Locker::calc_new_max_size(inode_t *pi, uint64_t size)
   uint64_t new_max = (size + 1) << 1;
   uint64_t max_inc = g_conf->mds_client_writeable_range_max_inc_objs;
   if (max_inc > 0) {
-    max_inc *= pi->get_layout_size_increment();
+    max_inc *= pi->layout.object_size;
```

Comment:
I agree this is a problem but I don't really think this is the right solution. The intent of the config is to limit the number of objects created for the writeable range (rather than a power of 2). However, the minimum number should at least be the stripe count as a simple write could allocate `stripe_count` objects.

Maybe restructure as:

````
uint64_t stripe_count = pi->layout.stripe_count;
uint64_t mds_client_writeable_range_max_inc_objs = g_conf->mds_client_writeable_range_max_inc_objs;
if (mds_client_writeable_range_max_inc_objs > 0) {
  // max_objects is a multiple of the stripe_count
  uint64_t max_objects = mds_client_writeable_range_max_inc_objs + (stripe_count-1);
  max_objects -= max_objects % stripe_count;
  new_max = std::min(new_max, size+max_objects*pi->layout.object_size);
}
````

Reply:
there is a ROUND_UP_TO a few line later, it does the job

Reply:
Ah, missed that. That works even better, thanks!